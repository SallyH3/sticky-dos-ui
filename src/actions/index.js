export const setCardList = cardList =>{
         console.log("VIP List", cardList)
         return {
           type: "SET_CARDLIST",
           cardList
         }};

export const deleteCard = (id) => ({
	type: 'DELETE_CARD',
	id
})