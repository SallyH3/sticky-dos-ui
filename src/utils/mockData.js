 export const mockCardList = [
  { 
    id: 0, 
    title: "Welcome to Sticky Do's", 
    content: [
      {
        id: 1,
        type: 'note',
        text: 'This is a standard note. Which is the default when you start typing in our input box ⤴️',
        checked: null
      }
    ]
  },
  { 
    id: 2, 
    title: "Example #2", 
    content: [
      {
        id: 1,
        type: 'list',
        text: 'This is a list item. If you click the ☑️ button in the input box',
        checked: false
      },
      {
        id: 2,
        type: 'list',
        text: 'You can add as many checkList items as you would like!',
        checked: true
      }
    ]
  }
]