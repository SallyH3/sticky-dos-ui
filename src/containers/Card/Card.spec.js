import React from 'react';
import { shallow } from 'enzyme';
import { Card, mapDispatchToProps } from './index.js';
import { cardList, deleteCard } from '../../actions';
import { NavLink } from 'react-router-dom';
import { mockCardList } from '../../utils/mockData.js';

const handleClick = jest.fn();

describe('Card', () => {
  let wrapper;
  let mockDeleteCard;

  beforeEach(() => {
    mockDeleteCard =jest.fn();
    wrapper = shallow(
      <Card 
        title={mockCardList[0].title}
        id={mockCardList[0].id}
        content={mockCardList[0].content}
        deleteCard={mockDeleteCard}
      />
    )
  })

  it('should match snapshot for note item', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot for list item', () => {
    wrapper=shallow(
      <Card 
        title={mockCardList[1].title}
        id={mockCardList[1].id}
        content={mockCardList[1].content}
      />
    )
    expect(wrapper).toMatchSnapshot();
  });

  it('should handleClick', () => {
    wrapper.instance().handleClick();
    expect(mockDeleteCard).toHaveBeenCalledWith(mockCardList[0].id);
  });

  describe('mapDispatchToProps', () => {

    it('should call dispatch when using a function from MDTP', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = deleteCard(0);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.deleteCard(0);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});