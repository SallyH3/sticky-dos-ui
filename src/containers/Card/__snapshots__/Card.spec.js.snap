// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot for list item 1`] = `
<article
  className="Card"
>
  <section
    className="Card__top"
  >
    <h4>
      Example #2
    </h4>
    <span
      className="Card__Btns"
    >
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-edit"
        />
      </button>
      <button
        className="Card__trash"
        onClick={[Function]}
      >
        <i
          className="far fa-trash-alt"
        />
      </button>
    </span>
  </section>
  <div
    className="content"
  >
    <p
      className="listItem"
      key="1"
    >
      <i
        className="far fa-square"
      />
      This is a list item. If you click the ☑️ button in the input box
    </p>
    <p
      className="listItem"
      key="2"
    >
      <i
        className="fas fa-check-square"
      />
      You can add as many checkList items as you would like!
    </p>
  </div>
</article>
`;

exports[`Card should match snapshot for note item 1`] = `
<article
  className="Card"
>
  <section
    className="Card__top"
  >
    <h4>
      Welcome to Sticky Do's
    </h4>
    <span
      className="Card__Btns"
    >
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-edit"
        />
      </button>
      <button
        className="Card__trash"
        onClick={[Function]}
      >
        <i
          className="far fa-trash-alt"
        />
      </button>
    </span>
  </section>
  <div
    className="content"
  >
    <p>
      This is a standard note. Which is the default when you start typing in our input box ⤴️
    </p>
  </div>
</article>
`;
